initial blueprint4 upgrade

veggiesaurus opened the pull request:
**Description**
Closes #1395. It may also close a number of table-related issues. 
Requires a fair amount of testing, and will likely break e2e tests. 


**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] changelog updated / ~no changelog update needed~
- [x] ~unit test added (for functions with no dependencies)~
- [x] ~API documentation added (for public variables and methods in stores)~

For dependencies:
- [x] ~e2e test passing~ / corresponding fix added / ~new e2e test created~
- [x] ~protobuf version bumped~ / no protobuf version bumped needed
- [x] ~protobuf updated to the latest dev commit~ / no protobuf update needed
- [x] ~corresponding ICD test fix added (`BackendService` changed)~ / no ICD test fix needed (`BackendService` unchanged)
- [x] ~user manual prepared (for large new features)~

veggiesaurus committed:
initial blueprint4 upgrade

veggiesaurus commented:
Assigning this to @kswang1029, as it is primarily a user-facing thing. Most of the upgrades were fairly straightforward. I've removed deprecated code where possible.

veggiesaurus committed:
Merge branch 'dev' into angus/1395_blueprint4

kswang1029 commented:
@veggiesaurus not sure why but I am seeing e2e test failures due to changes of xpath (eg. the file filter text field). I will need more time to investigate this. And you are right about the color theme: the contrast is increased and the color theme appears less warm.

kswang1029 commented:
One issue: the initial file browser highlight box is squeezed. 
with this PR
![Screen Shot 2022-09-01 at 20 56 31](https://user-images.githubusercontent.com/20819712/187919116-c260b690-4c4d-4a2e-b54b-b863575a1423.png)
current dev
![Screen Shot 2022-09-01 at 20 56 54](https://user-images.githubusercontent.com/20819712/187919120-cd396efb-b1c6-4d04-a743-ca514c301e58.png)

kswang1029 commented:
in the file browser, the sorting function in the the Filename, Type, Size, and Date columns does not work anymore.

kswang1029 commented:
@veggiesaurus @YuHsuan-Hwang there are a number of issues due to this library upgrade. I will summarize them in a google doc, just for better readability than posting here as a comment thread.

kswang1029 commented:
@veggiesaurus please refer to https://docs.google.com/document/d/1UgbYN7CzW2ziPeYo7bn_vDxHrSgs1ri6QDjXS86ylYk/edit?usp=sharing for a list of issues I observed with this blueprintjsV4 upgrade. Basically there are major issues with 
- spinbox
- html table
- blueprint table 🤦🏻‍♂️

TienHao committed:
fixed blueprintjs upgrading from v3 to v4 issues

TienHao committed:
fixed upgrading issues

- the misplaced lock icon in Contour dialogue

- Table2 rerender issues in Cursor Info and hypercube dialogue

TienHao committed:
Merge branch 'dev' into angus/1395_blueprint4

TienHao committed:
Merge branch 'dev' into angus/1395_blueprint4

TienHao committed:
merge dev

TienHao committed:
Merge dev branch

TienHao committed:
update css tag with bp3 to bp4 in other commits

TienHao committed:
update changelog

TienHao committed:
Merge dev branch and replace deprecated v3 imports and bp3 tag

kswang1029 commented:
@TienHao most of the issues in https://docs.google.com/document/d/1UgbYN7CzW2ziPeYo7bn_vDxHrSgs1ri6QDjXS86ylYk/edit are addressed. 👍 
there are some new others. Please refer to the above doc.

TienHao committed:
Merge dev branch and replace bp3 to bp4 tags in scss files

TienHao committed:
fixed filter input unfocused issue, caused by focus box workaround

TienHao committed:
modified menu item style

TienHao committed:
removed redundant menu divider from branch merge

kswang1029 committed:
fix widgets menu styling regression

TienHao committed:
fix widgets menu styling

TienHao committed:
revert fix widgets menu stlying commit

TienHao committed:
merge dev

TienHao committed:
merge label_styling_regression branch

TienHao committed:
merge dev

TienHao committed:
merge dev

TienHao committed:
merge dev branch and remove bp3 css tag

TienHao committed:
merge dev and fix conflicts

YuHsuan-Hwang commented:
Upgrade to blueprint v5 in the next release (v5-beta).

TienHao committed:
merge dev branch and resolve conflicts

* update package-lock

TienHao committed:
remove fill in the popoverProps of Select2

TienHao committed:
replace bp3 to bp4 in css tags

TienHao committed:
remove non-required v4 package of ResizeObserver

TienHao committed:
modified test for role names of MenuDivider and MenuItem2 has changed after v4

TienHao committed:
modified menu items check

TienHao committed:
merge dev branch

TienHao committed:
replaced bp4 css tag to bp5

TienHao committed:
initial blueprintjs v5

* updated packages

* replaced Popover2, MenuItem2, Tooltip2, Select2...

* replaced popover2 css tag to popover

* removed deprecated Ixxx props

* fixed MenuItem text prop required

* fixed Icon

TienHao committed:
modified test for role name changed

TienHao committed:
replaced blueprintjs Table to Table2

TienHao committed:
updated packaged-lock.json for Blueprintjs v5

TienHao committed:
replaced bp5 with ns variable in SCSS

TienHao committed:
use proper class names from the Classes import

TienHao committed:
merge dev branch

TienHao committed:
merged dev branch and solved conflict

TienHao committed:
update package and changelog

TienHao committed:
minor refactoring

TienHao committed:
UI modification

* FileInfo header searching button misplacement fix

* added back fill propperty in selector of data source

TienHao committed:
fix UI misplacement in Fitting Dialog after version upgrade

TienHao committed:
info string style adjustment

TienHao committed:
adjusted selected background color of Region List

TienHao committed:
merge dev branch and solve conflicts

TienHao committed:
adjusted scaling selector style and removed legacy code

TienHao committed:
removed comment

YuHsuan-Hwang commented on src/App.scss:
```diff
@@ -58,11 +58,17 @@ body {
     margin-right: -1px !important;
 }
 
+// override menu item style with BlueprintJS v3 style
```
Remove comments.

TienHao replied:
removed!

YuHsuan-Hwang reviewed (changes requested):
The changes look nice. Thanks for all the hard work!

One minor issue: the fitting dialog right-bottom toolbar position needs adjustment.
<img src="https://github.com/CARTAvis/carta-frontend/assets/43841102/df3c404a-9564-45af-9b58-a4454e5e23ef" width="400" />

TienHao committed:
merge dev branch

TienHao committed:
adjust css of fitting result buttons

TienHao committed:
replaced deprecated Overlay with Overlay2 for React 18 strict mode

TienHao committed:
adjust file browsser UI

TienHao committed:
fixed typo

TienHao committed:
fixed missing help buttons in dialog and misplaced focus box

* refactoring code with ref

TienHao committed:
merge dev branch

TienHao committed:
adjust column widths

TienHao committed:
create PreferenceSilentSettings to store all the preferences not in the preference dialog

TienHao committed:
added backdrop for help drawer to prevent drawer reopening on help buttons

TienHao committed:
merge dev branch and solve conflicts

TienHao committed:
merge dev branch and solve conflicts in package-lock.json

TienHao committed:
solve conflicts and merge dev branch

TienHao committed:
merge dev branch

TienHao committed:
fix no update in file list when filtering result have same number of files

kswang1029 reviewed (approved):
Well done @TienHao 👍

TienHao committed:
revert commit 9ac13b059

YuHsuan-Hwang reviewed (approved):
New changes look good.
